src/sig_stfl/xmss/sig_stfl_xmss_secret_key_functions.c

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: (Apache-2.0 OR MIT) AND CC0-1.0
+
+#include <oqs/oqs.h>
+#include <string.h>
+#include <stdbool.h>
+#include "sig_stfl_xmss.h"
+
+#if defined(__GNUC__) || defined(__clang__)
+#define XMSS_UNUSED_ATT __attribute__((unused))
+#else
+#define XMSS_UNUSED_ATT
+#endif
+
+extern inline OQS_SIG_STFL_SECRET_KEY *OQS_SECRET_KEY_XMSS_new(size_t length_secret_key) {
+
+	// Initialize the secret key in the heap with adequate memory
+	OQS_SIG_STFL_SECRET_KEY *sk = malloc(sizeof(OQS_SIG_STFL_SECRET_KEY));
+	if (sk == NULL) {
+		return NULL;
+	}
+	memset(sk, 0, sizeof(OQS_SIG_STFL_SECRET_KEY));
+
+	sk->length_secret_key = length_secret_key;
+
+	// Secret serialize/deserialize function
+	sk->serialize_key = OQS_SECRET_KEY_XMSS_serialize_key;
+	sk->deserialize_key = OQS_SECRET_KEY_XMSS_deserialize_key;
+
+	// Initialize the key with length_secret_key amount of bytes.
+	sk->secret_key_data = (uint8_t *)malloc(sk->length_secret_key * sizeof(uint8_t));
+
+	if (sk->secret_key_data == NULL) {
+		OQS_MEM_insecure_free(sk);
+		return NULL;
+	}
+
+	memset(sk->secret_key_data, 0, sk->length_secret_key);
+
+	// Set application specific context
+	sk->context = NULL;
+
+	// Point to associated OQS_SIG_STFL object
+	sk->sig = NULL;
```

Comment:
Shouldn't this be set since a OQS_SIG_STFL object is required to generate a secret key?

Reply:
It looks to me like the `->sig` field is never actually used (other than being set to NULL in this one function). I'm not sure whether we should get rid of it entirely. Tagging @ashman-p and @cothan for input.

Reply:
Hi @SWilson4, you are correct. it turns out that the member is not used and can be removed.

Reply:
I've removed the `sig` member with the latest commit.